Source/Core/DolphinQt2/GCMemcardManager.h

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <memory>
+
+#include <QDialog>
+
+#include "Common/MsgHandler.h"
+#include "Core/HW/GCMemcard/GCMemcard.h"
```

Comment:
Not possible with a ``unique_ptr``

Reply:
`unique_ptr` allows forward declarations if you get an error, it's related to something else.

Reply:
It's because the destructor for the memcard manager is being inlined, and because that, it's only seeing the incomplete type. If you default the destructor in the cpp file:

```cpp
// header
class blah
{
public:
  ~blah();
};

// cpp
blah::~blah() = default;
```

it should fix that issue.

Reply:
Yeah figured that out now aswell. Thanks for your patience.